src/Select/Select.js

```diff
@@ -1,23 +1,38 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import classnames from 'classnames';
 
 import './Select.scss';
 
 export default function Select({
-  name, label, options, children,
+  name, valueAttr = 'value', textAttr = 'text', label, options, children, placeholder, value, error, ...rest
 }) {
   if (children) {
     return <div className="dfo-select">{children}</div>;
   }
+
+  const className = (value !== '') ? undefined : 'displayPlaceholder';
+
   return (
-    <div className="dfo-select">
-      {label && <label htmlFor={`select-${name}`}>{label}</label>}
-      <select id={`select-${name}`} name={name}>
-        {options.map(option => (
-          <option key={option}>{option}</option>
-        ))}
+    <label
+      htmlFor={`select-${name}`}
+      className={classnames('dfo-select', { 'dfo-select--is-error': error })}
+    >
+      <div
+        className="dfo-select dfo-error-wrapper"
+        aria-live="assertive"
+        aria-relevant="additions removals"
+      >
+      {label}
+      <select aria-invalid={!!error} id={`select-${name}`} name={name} value={value} className={className} {...rest}>
+        { (placeholder || placeholder === '') && <option key={placeholder} value='' disabled>{placeholder}</option> }
+        { options.map(option => {
+            return (typeof option === 'object') ? <option key={option[valueAttr]} value={option[valueAttr]}>{option[textAttr]}</option> : <option key={option}>{option}</option>;
```

Comment:
I see you've added a configurable options object where you can define key and text values. 

Have you looked at [Object.entries()](https://devdocs.io/javascript/global_objects/object/entries)? With that something like this might be possible to do. 

```javascript
{Object.entries(options).map(([key, value]) => {
            return (
              <option key={key} value={key}>
                {value}
              </option>
            );
})}
```

I see that you also want to support a plain array of options like `['alfa', 'beta', 'gamma']`. Could you  express that by expanding the PropTypes for the options object so that the array contains either objects or plain strings?